src/main/java/reactor/ipc/netty/NettyContext.java

```diff
@@ -125,5 +246,33 @@ default NettyContext addDecoder(ChannelHandler handler){
 	 *
 	 * @return {@literal this}
 	 */
-	NettyContext onClose(Runnable onClose);
+	default NettyContext onClose(Runnable onClose){
+		onClose().subscribe(null, e -> onClose.run(), onClose);
+		return this;
+	}
+
+	/**
+	 * Remove a named handler if present and return this context
+	 *
+	 * @param name handler name
+	 *
+	 * @return this NettyContext
+	 */
+	default NettyContext removeHandler(String name) {
+		ReactorNetty.removeHandler(channel(), name);
+		return this;
+	}
+
+	/**
+	 * Replace a named handler if present and return this context.
+	 * If handler wasn't present, an {@link RuntimeException} will be thrown
+	 *
```

Comment:
Add a note "note that if your handler is of different type, there could be an extraneous 'extractor' still in place" so you should avoid this kind of replacement" or something like that?